### PR TITLE
Expose clearButtonMode RX.TextInput property

### PR DESCRIPTION
**Pull request by reseul**
This piggybacks on existing iOS property, and the UWP implementation will follow suit.

### CODE REVIEW

File: docs/docs/components/textinput.md [hunk 1]
```diff
@@ -117,6 +114,11 @@ secureTextEntry: boolean = false;
 // Should spell checking be applied to contents?
 spellCheck: boolean = [value of autoCorrect];
 
+// iOS and Windows only property for controlling when the clear button should appear on the right side of the text view.
+// Default behavior is dependent on platform: equivalent to 'never' on iOS, and 'always' on Windows.
```
Comment (hnager) [inline docs/docs/components/textinput.md]: Would it be so bad to have the default be consistent? We could explicitly set 'never' if the incoming prop is undefined in nativecommon/TextInput

Comment (reseul) [reply docs/docs/components/textinput.md]: We could, it's just that I'm torn. The default UWP behavior is something Windows users got accustomed to.

Comment (hnager) [reply docs/docs/components/textinput.md]: I don't feel strongly about it... it's just an implementation detail that a dev needs to know and think about.

File: docs/docs/components/textinput.md [hunk 2]
```diff
@@ -117,6 +114,11 @@ secureTextEntry: boolean = false;
...
+// Button is hidden when text view is empty, regardless of the property value.
+clearButtonMode: 'never' | 'while-editing' | 'unless-editing' | 'always'
```
Comment (erictraut) [inline docs/docs/components/textinput.md]: Nit: missing a semicolon at the end.

Comment (erictraut) [reply docs/docs/components/textinput.md]: We could define another value 'default' and map that to the default value for each platform. Or we could just specify that an undefined value maps to the default behavior. I'm fine with either approach.

Comment (reseul) [reply docs/docs/components/textinput.md]: Howard would prefer the default to be 'never', aligned with the S4L design that abhors the clear buttons.
I'm more inclined to give platform some "personality" choice..
Currently it is coded as/assumes "undefined means whatever platform chooses as the default"

File: docs/docs/components/textinput.md (hunk 1 above)
Comment (mschmuckler) [reply docs/docs/components/textinput.md]: I'm nearly done with the react-native-windows implementation of it. Default mode (if no value is provided) will be the behavior as it is right now, so nothing to worry about for devs who aren't aware of this prop. I've also ensured the 4 modes behave exactly like their iOS counterparts.

**Pull request merged**